src/host/polyfills/simpleView.test.ts

```diff
@@ -2,11 +2,31 @@
 // Licensed under the MIT License.
 
 describe("simpleView", () => {
+    it("revealInVSCode calls openInEditor", async () => {
+        const apply = await import("./simpleView");
+        const expected = {
+            columnNumber: 0,
+            lineNumber: 0,
+            uiSourceCode: {
+                _url: "http://bing.com",
+            },
+        };
+        const mockOpen = jest.fn();
+        (global as any).InspectorFrontendHost = {
+            openInEditor: mockOpen,
+        };
+
+        await apply.revealInVSCode(expected, false);
+
+        expect(mockOpen).toHaveBeenCalled();
+    });
+
     it("applyCommonRevealerPatch correctly changes text", async () => {
         const apply = await import("./simpleView");
         const result = apply.applyCommonRevealerPatch(
             "Common.Revealer.reveal = function(revealable, omitFocus) { // code");
-        expect(result).toEqual("Common.Revealer.reveal = function() { Promise.resolve(); return; // code");
+        expect(result).toEqual(
+            expect.stringContaining("Common.Revealer.reveal = function revealInVSCode(revealable, omitFocus) {"));

```

Comment:
I'm a little confused on this, can you help explain what is happening? I would think expect.stringContaining() to return a boolean, and I thought result would be a string, so how would they be equal? Or is result a boolean as well?

Reply:
These are special Jest matchers that can be used instead of a literal value. See: https://jestjs.io/docs/en/expect.html#expectstringcontainingstring

You can read it as "expect the result to equal a string containing blah"

Reply:
That makes sense, my question is: won't expect.stringContaining() return a promise that resolves to pass/fail? and then you are expecting result (which i assume is a string) to equal that promise? Or does expect.stringContaining() actually return the string instead of pass/fail?

Reply:
I'm not 100% sure how the internals of Jest are implemented, but stringContaining() is not a comparison function like the ones that start with 'to'. It actually returns an object (a jest asymmetric matcher) which Jest treats like a literal value when it is chained together with the toEqual call. This lets Jest perform the fuzzy matching logic with the actual value in `result`.

Reply:
Cool! Thanks for explaining that.